enhancements/update/targeted-update-edge-blocking.md

```diff
@@ -0,0 +1,400 @@
+    For example, some issues may be races where the impacted cluster set is a random subset of the vulnerable cluster set.
+    Any targeting of the blocked edges will reduce the number of blocked clusters which would have not been impacted, and thus reduce the contention between protecting vulnerable clusters and inconveniencing invulnerable clusters.
+* Specifying a particular update service implementation.
+    This enhancement floats some ideas, but the details of the chosen approach are up to each update service's maintainers.
+
+## Proposal
+
+### Enhanced graph-data schema for blocking edges
+
+[The blocked-edges schema][block-edges] will be extended with the following new properties:
+
+* `url` (optional, [string][json-string]), with a URI documenting the blocking reason.
+    For example, this could link to a bug's impact statement or knowledge-base article.
+* FIXME: [also reason/message?](#metadata-to-include-when-blocking-a-conditional-request)
+* `clusters` (optional, [object][json-object]), defining the subset of affected clusters.
+  If any `clusters` property matches a given cluster, the edge should be blocked for that cluster.
+  If `clusters` is unset, the edge is blocked for all clusters.
+  * `promql` (optional, [string][json-string]), with a [PromQL][] query describing affected clusters.
+    This query will be evaluated on the local cluster, so it has access to data beyond the subset that is [uploaded to Telemetry][uploaded-telemetry].
+    The query should return a 0 if the update should be allowed and a 1 if the update should be blocked.
+  * FIXME: [`platform` or other properties that are easier to evaluate than PromQL?](#non-promql-filters)
+
+[The schema version][graph-data-schema-version] would also be bumped to 1.1.0, because this is a backwards-compatible change.
+Consumers who only understand graph-data schema 1.0.0 would ignore the `clusters` property and block the edge for all clusters.
+The alternative of failing open is discussed [here](#failing-open).
+
+### Enhanced Cincinnati JSON representation
+
+[The Cincinnati graph API][cincinnati-api] will be extended with a new top-level `conditionalEdges` property, with an array of conditional edge [objects][json-object] using the following schema:
+
+* `from` (required, [string][json-string]), with the `version` of the starting node.
+* `to` (required, [string][json-string]), with the `version` of the ending node.
+* `qualifiers` (optional, [array][json-array], with qualifications around the recommendation.
+  Each entry is an [object][json-object] with the following schema:
+  * `url` (optional, [string][json-string]), with a URI documenting the issue, as described in [the blocked-edges section](#enhanced-graph-data-schema-for-blocking-edges).
+  * FIXME: [also reason/message?](#metadata-to-include-when-blocking-a-conditional-request)
+  * `clusters` (optional, [object][json-object]).
```

Comment:
this name is slightly weird for how it is used.......  would `blockedClusters` help clarify?

we are after all defining an edge.  Clusters is indicating the clusters this edge does NOT apply to.

Reply:
The release image itself defines the edge, by declaring an initial set of versions which can be updated to it.  These `blocked-edges/...` entries prune that list of recommendations.  This new `clusters` property limits the scope of that blocking/pruning to specific clusters.  I'm personally fine with inheriting that context from the `blocked-edges/` directory name, like we do with the existing `from` and `to`, but I'm not set against `blockedClusters` or similar, if folks think the closer echo is useful.  Thoughts?

Reply:
i think fundamentally it would be useful to always say "conditionallyBlocked[edges]" or "conditionallyAllowed[edges]" anytime we are using "conditional" in this api.  And then the "clusters" query defines the condition under which the thing is blocked(or allowed).

Alternatively, if we call them "conditionalEdges" then i'd expect the fields within that struct to indicate what influence they have on the condition, e.g. "allowConditions" and "blockConditions" or just "allow" and "block" (in case you want to default to allow vs default to block)